### PR TITLE
Create NTA upload worker (UA-655)

### PR DESCRIPTION
**Pull request by kotodharma**
Seems to be working, although I was having some browsery issues in testing, but since everything in the back end seems fine, and reloading the pages showed UI stuff was somehow right. If there are dynamic UI issues, we can deal with them, because I need to get this out the door :=)

Before testing, create a root category for NTA, something like this `Category.create(universe: 'NTA', name: 'NTA Data Portal', ancestry: nil)`

Unfortunately there's a lot of cruft in the diffs, esp in `app/models/nta_upload.rb` and others for which an older version of the file was checked in at the previous story.

As for testing, I think you know the drill pretty well :=)

### CODE REVIEW

File: app/helpers/categories_helper.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def show_table(root, first, last)
           '</li>'+"\n"
     end
 
-    categories = (root.children.to_a).sort_by!{ |cat| cat.list_order }
+    categories = (root.children.to_a).sort_by!{ |cat| cat.list_order || cat.name }  ## sort by alpha if list_order is nil -dji
```
Comment (jonpage) [inline app/helpers/categories_helper.rb]: When I test this with a mixed array with some items having a nil list_order, I get the following error:
```
ArgumentError: comparison of Fixnum with String failed
```

Comment (kotodharma) [inline app/helpers/categories_helper.rb]: I recognized this problem, but assumed that for any given universe, `list_order` would either be all filled in, or all null. Probably not a great assumption to make. Reverted.

**Comment by jonpage** [comment]
Looks like I managed to leave off the DATA_PATH environment from `rails s.` It would be helpful if we had better alerting of missing environmental variables on startup.

**Comment by jonpage** [comment]
The upload process took about 12 minutes

**Pull request merged**